### ISSUE TITLE
Adding a merged csv file download option to the resource page

### ISSUE THREAD

**Issue by kbraak** [gbif/ipt#1165]
In the case of multiple mapping the consumer of the Darwin Core Archive needs to merge (outer join) the multiple csv data files for having all the mapped data in one spreadsheet. I assume a spreadsheet application is being used. I have found no "Darwin Core Archive Viewer".
A new download option would be a zip file which includes a merged file and RDF (or also EML file).
This new format could be harvested and be presented also in GBIF web site, even though the format is not standard.

For now, I can do it myself each time a new resource version is being published, for example:
csvjoin -c "id,id" --outer occurrence.txt measurementorfact.txt > data.csv

Best regards,
Menashè

**Comment by kbraak** [comment]
Thanks for reporting this issue, it's under consideration.

**Comment by meliezer** [comment]
Actually with the new event core and the option to have a hirerarchy of events, the end user can have for example event core with Occurrence extension, MoF extension and a big tree of events, while some properites, like coordinates, can be found in an arbitary level of the events tree. A script has no problem get all the information, but a human using Excel?... So, the feature can be named "Include all metadata in one big Excel file".

Best regards,
Menashè

**Comment by meliezer** [comment]
I haven't mentioned that of course this csv/tsv should not include empty columns.

Best regards,
Menashè

**Comment by dvdscripter** [comment]
Here on Brazil we're starting to use Event Core and extensions for ecological data. Researchers are complaining of the lack of possibility to format data to original format without experts in spreadsheet.

We're afraid to lose sources of ecological data (where we find many environment variables). On TDWG mail list we have a ongoing discussion on the matter.

**Comment by meliezer** [comment]
I understand: data providers problem, not data consumers. I may look later
also at the TDWG mailing list.
A problem of who provides the data, not for IPT resource manager which
needs to map the terms, right?
If you provide an example, maybe there is a solution.
You may already know that one worksheet file can be used in multiple
mapping.
You can uncheck for a specific mapping the auto-mapped columns you don't
need.

--
Menashè


2015-10-27 20:26 GMT+01:00 David Valentim Dias <notifications@github.com>:

**Comment by dvdscripter** [comment]
No, this feature request is a quality of life improvement very needed now.
How many researchers or undergrads you known will make a simples analysis
using ecological data divided in three relational tables?
I'm not saying this is impossible to achieve but turn people who might use
this data away.
Also from discussion on mailing list I think we have two options: create a
verbatim containing the layout of columns; or create a link to download the
original data (or merged as above issue) from IPT.

2015-10-27 19:11 GMT-02:00 meliezer <notifications@github.com>:

> I understand: data providers problem, not data consumers. I may look later
> also at the TDWG mailing list.
> A problem of who provides the data, not for IPT resource manager which
> needs to map the terms, right?
> If you provide an example, maybe there is a solution.
> You may already know that one worksheet file can be used in multiple
> mapping.
> You can uncheck for a specific mapping the auto-mapped columns you don't
> need.
>
> --
> Menashè
>
>
> 2015-10-27 20:26 GMT+01:00 David Valentim Dias <notifications@github.com>:
>
> > Here on Brazil we're starting to use Event Core and extensions for
> > ecological data. Researchers are complaining of the lack of possibility
> to
> > format data to original format without experts in spreadsheet.
> >
> > We're afraid to lose sources of ecological data (where we find many
> > environment variables). On TDWG mail list we have a ongoing discussion on
> > the matter.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/gbif/ipt/issues/1165#issuecomment-151617305>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gbif/ipt/issues/1165#issuecomment-151645302>.
>



-- 



[image: ass_sibbr]

*David Valentim Dias*

*Biodiversity Data Management*

SiBBr | MCTI | PNUMA (UNEP)

Phone: +55 61 3329 6045
Phone: +55 61 9359 6151
Skype: ctenidae

dvdias@sibbr.gov.br

www.sibbr.gov.br

**Comment by meliezer** [comment]
Hi David,
I'm pleased that you and others agree about the need for downloading a
merged file.
However, the option of downloading the source files is not good enough.
You assume there is only one source file, and that the column names are the
standrad names (To be used by auto-mapping of IPT).
However, one can have for example two files: abundance and biomass.
In my case, even more source files:  A Java program creates separate files
for each mapping, as source files to be mapped by the IPT manager.

--
Menashè


2015-10-28 12:46 GMT+01:00 David Valentim Dias <notifications@github.com>:

**Comment by meliezer** [comment]
Let me rephrase my wish request.
IPT change should be limited to:
1. IPT settings: Add an option to set the web address of a DwC2Matrix web service. More details below. Ideally, this web service should be also a part of the IPT installation.
2. During the publication IPT will submit the created DwC-A to the above web service using its API, and get a worksheet file to be saved on disk. IPT will offer the user the option to download a resource as a matrix, offering this file (.ods) which can be used by speardsheet applications, e.g. Excel.

I would add “Download as matrix” also to GBIF web site, add possibly also a rich web service client which I mention below.

DwC viewer and exporter web service
DwC is efficient for storing information, but difficult to handle by consumers and analysis applications.
This web service should be able to translate to one big matrix even the most complex DwC file in which merging is not enough, since there is a parentEventID and a hirerarchy of even more than three levels. Empty columns will not be included in the resulted matrix.
The web service can offer also exporting as an OBIS scheme matrix.
Basically, the Java code can use org.apache.poi for reading the tables inside the DwC file, staring from the core and adding the missing columns taking the information from the extension tables, using for example occurrenceID which connects Occurrence core and MoF extension.
The project should also include an example of a client to this web service: A web page which offers paging and many of a spreadsheet functions: selecting which columns should be shown/hidden, advanced filter, filter by keyword, pivoting, etc. Presenting also the EML information to end user can be offered by the client.
I'm sorry I don't have time for developing this web service as a new Github project, but I can continue providing ideas and feedback. Ideally, GBIF could develop this web service.
Also a desktop application can be developed, but a web service is better. A desktop application can be a client of the web service.

--
Menashè

**Comment by kbraak** [comment]
Please note the [CSV on the Web Working Group](http://www.w3.org/2013/csvw/) aims to produce a new standard format for tabular datasets comprising single or multiple files using CSV, or similar, format. GBIF submitted DwC-A as a [use case](http://w3c.github.io/csvw/use-cases-and-requirements/index.html#UC-PublicationOfBiodiversityInformation) for consideration by the working group. In lieu of the fact that a new W3C  format for tabular data may replace DwC-A in the future, I prefer not introducing another download format to the IPT at this time.

If you or others require this service I would encourage you to work together to collaboratively develop it. Thanks.
